bin/deploy.sh

```diff
@@ -12,146 +12,149 @@ THIS_DIR="$(cd "$(dirname -- "$0")"; pwd)"
+  echo "### COMPLETED ###"
+
+  # set paths in env file to be correct inside container
+  echo "### Exporting Service Account Keys: $SERVICE_ACCOUNT_JSON_PATH, $READ_ONLY_SERVICE_ACCOUNT_JSON_PATH ###"
+  echo "export SERVICE_ACCOUNT_KEY=/home/app/webapp/config/$SERVICE_ACCOUNT_FILENAME" >> $CONFIG_FILENAME
+  echo "export READ_ONLY_SERVICE_ACCOUNT_KEY=/home/app/webapp/config/$READ_ONLY_SERVICE_ACCOUNT_FILENAME" >> $CONFIG_FILENAME
+  echo "### COMPLETED ###"
+
+  # init folder/repo if this is the first deploy on this host
+  $SSH_COMMAND "if [ ! -d $DESTINATION_BASE_DIR ]; then sudo mkdir -p $DESTINATION_BASE_DIR && sudo chown -R $SSH_USER: $DESTINATION_BASE_DIR; fi"
+  run_remote_command "if [ ! -d .git ]; then sudo rm -rf ./* && git clone $SCP_REPO .; fi"
+
+  # move secrets to remote host
+  echo "### migrating secrets to remote host ###"
+  copy_file_to_remote ./$CONFIG_FILENAME $PORTAL_SECRETS_PATH || exit_with_error_message "could not move $CONFIG_FILENAME to $PORTAL_SECRETS_PATH"
+  copy_file_to_remote ./$SERVICE_ACCOUNT_FILENAME $SERVICE_ACCOUNT_JSON_PATH || exit_with_error_message "could not move $SERVICE_ACCOUNT_FILENAME to $SERVICE_ACCOUNT_JSON_PATH"
+  copy_file_to_remote ./$READ_ONLY_SERVICE_ACCOUNT_FILENAME $READ_ONLY_SERVICE_ACCOUNT_JSON_PATH || exit_with_error_message "could not move $READ_ONLY_SERVICE_ACCOUNT_FILENAME to $READ_ONLY_SERVICE_ACCOUNT_JSON_PATH"
+  echo "### COMPLETED ###"
+
+  # update source on remote host to pull in changes before deployment
+  echo "### pulling updated source from git on branch $GIT_BRANCH ###"
+  run_remote_command "git remote update" || exit_with_error_message "git remote update failed"
+  run_remote_command "git fetch --all --tags" || exit_with_error_message "git fetch --all failed"
+  run_remote_command "git checkout yarn.lock" || exit_with_error_message "could not reset yarn.lock file"
+  run_remote_command "git checkout $GIT_BRANCH && git pull" || exit_with_error_message "could not checkout $GIT_BRANCH"
+  echo "### COMPLETED ###"
+
+  # if this is a production deployment (or a hotfix deployment on staging), get the latest tag to pass to remote_deploy
+  if [[ $PASSENGER_APP_ENV == "production" ]] || [[ $HOTFIX == "true" ]]; then
+    VERSION_TAG=$(extract_release_tag 0)
+  fi
+
+  if [[ $ROLLBACK == "true" ]]; then
+    # checkout the requested tag (usually current release - 1)
+    echo "### ROLLING BACK DEPLOYMENT BY $TAG_OFFSET RELEASE TAG ###"
+    echo "### determining requested release rollback tag on $GIT_BRANCH ###"
+    ROLLBACK_TAG=$(extract_release_tag $TAG_OFFSET) || exit_with_error_message "could not get rollback tag in $GIT_BRANCH"
+    echo "### rolling back release to tag: $ROLLBACK_TAG on $GIT_BRANCH ###"
+    run_remote_command "git checkout tags/$ROLLBACK_TAG" || exit_with_error_message "could not checkout tags/$ROLLBACK_TAG on $GIT_BRANCH"
+    VERSION_TAG="$ROLLBACK_TAG"
     echo "### COMPLETED ###"
+  fi
```

Comment:
Per your standup comment, it seems the effective changes are limited to this section, with the other changes are less relevant indentation refinements.